src/hotspot/cpu/x86/stubRoutines_x86.cpp

```diff
@@ -227,6 +227,43 @@ juint StubRoutines::x86::_shuf_table_crc32_avx512[] =
 };
 #endif // _LP64
 
+jint StubRoutines::x86::_arrays_hashcode_powers_of_31[] =
```

Comment:
This should be declared only for LP64.

Reply:
Hmm, I guess same goes for all the new `arrays_hashcode` methods in `c2_MacroAssembler_x86`, since we only wire this up properly on 64-bit. I'll make a pass to put `_LP64` guards around all new methods.